File: CoteraWashington/src/main/java/day_02/PartTwo.java
package day_02;

import static day_02.Common.me;
import static day_02.Common.op;

public class PartTwo {

    //    R 	A 1 X Lose
    //    P     B 2 Y Draw
    //    S	    C 3 Z Win

    public static int myPlayScore(String op, String result) {
        return switch (op) {
            case "A" -> switch (result) {
                case "X" -> 3;
                case "Y" -> 4;
                default -> 8;
            };
            case "B" -> switch (result) {
                case "X" -> 1;
                case "Y" -> 5;
                default -> 9;
            };
            default -> switch (result) {
                case "X" -> 2;
                case "Y" -> 6;
                default -> 7;
            };
        };
    }

    public static void main(String[] args) {

        int score = 0;

        for (int i = 0; i < op.size(); i++) {

            String m = me.get(i);
            String o = op.get(i);
            score += myPlayScore(o, m);

        }

        System.out.println(score);
    }
}

File: GoussasAlexander/lib/Aoc/Day/Two/PartTwo.hs
module Aoc.Day.Two.PartTwo where

import           Aoc.Class
import           Aoc.Day.Two.Common

import           Data.Proxy
import           Data.Text          (Text)

data Day2Part2

data RoundResult = Win | Lose | Draw deriving Show

parseRoundResult :: Text -> RoundResult
parseRoundResult "X" = Lose
parseRoundResult "Y" = Draw
parseRoundResult "Z" = Win
parseRoundResult c   = error $ "No parse for char " <> show c

calculateHandForResult :: Hand -> RoundResult -> Hand
calculateHandForResult Paper Win     = Scissors
calculateHandForResult Paper Lose    = Rock
calculateHandForResult Paper Draw    = Paper
calculateHandForResult Scissors Win  = Rock
calculateHandForResult Scissors Lose = Paper
calculateHandForResult Scissors Draw = Scissors
calculateHandForResult Rock Win      = Paper
calculateHandForResult Rock Lose     = Scissors
calculateHandForResult Rock Draw     = Rock

day2Part2 :: Proxy Day2Part2
day2Part2 = Proxy

instance Aoc (Proxy Day2Part2) where
  type Result (Proxy Day2Part2) = Int

  solution _ input =
    let rounds = parseInputWith parseRoundResult input
        rounds' = map (\(a, b) -> (a, calculateHandForResult a b)) rounds
     in pure $ sum $ map calculateScoreForRound rounds'
